### ISSUE TITLE
TypeVar special handling doesn't recognize ** for ARG001

### ISSUE THREAD

**Issue by henryiii** [astral-sh/ruff#22178]
### Summary

This example produces a false positive:

```python
import typing

def f(
    *args: object,
    default: object = None,  # noqa: ARG001
    **kwargs: object,
) -> None:
    typing.TypeVar(*args, **kwargs)

__all__ = ["f"]
```

```console
$ ruff check --isolated --select=ARG tmp.py
ARG001 Unused function argument: `kwargs`
 --> tmp.py:6:7
  |
4 |     *args: object,
5 |     default: object = None,  # noqa: ARG001
6 |     **kwargs: object,
  |       ^^^^^^
7 | ) -> None:
8 |     typing.TypeVar(*args, **kwargs)
  |

Found 1 error.
```

To handle Python 3.13's new `default=` argument to `TypeVar`, a pattern including the above is helpful. See https://github.com/scikit-build/scikit-build-core/pull/1202.

**Comment by ntBre** [comment]
Thanks for the report! It looks like we indeed don't visit variadic keyword arguments at all in `TypeVar`s, which seems like a bug for sure.

https://github.com/astral-sh/ruff/blob/e3498121b4700f5a9dce9bf6d72af4bc411e262d/crates/ruff_linter/src/checkers/ast/mod.rs#L1863-L1877

I think we should visit this, and possibly some of  our other `typing::Callable` variants here, as `non_type_definition`s so that references to their bindings are actually recorded. This seems analogous to our change for `NamedTuple` in https://github.com/astral-sh/ruff/pull/8810.

**Comment by ValdonVitija** [comment]
@ntBre Is anyone working on this, if not I would for sure like to give this one a try!

**Comment by ntBre** [comment]
Not that I know of, go for it!
